ingesters/preprocessors/preprocessors.md

```diff
@@ -43,6 +43,74 @@ Log-Level=INFO
 
 This configuration defines two data consumers (Simple Relay calls them "Listeners") named "default" and "syslog". It also defines a preprocessor named "timestamp". Note how the "default" listener includes the option `Preprocessor=timestamp`. This specifies that entries coming from that listener on port 7777 should be sent to the "timestamp" preprocessor. Because the "syslog" listener does not set any `Preprocessor` option, entries coming in on port 601 will not go through any preprocessors.
 
+### Testing Preprocessors
+
+The [preprocessortest](https://github.com/gravwell/gravwell/tree/main/tools/preprocessortest) program is an open source tool to provide a simple scaffolding for testing ingest preprocessor stacks, it is designed to accept a data export from Gravwell and run the raw data through a set of preprocesors without actually ingesting any data.
+
+#### Getting Started
+
+First you will need to get a raw data export from some unprocessed data, this can be a simple text file that is line delimited or a JSON export of data from Gravwell.  For example, if we were working with syslog data from the `syslog` tag we might run the following query:
+
+```
+tag=syslog limit 1000 | raw
+```
+
+#### Example Test Config
+
+An example stack of preprocessors may have a configuration file like so:
+
+```
+[Global]
+	Preprocessor = apprtr
+	Preprocessor=loginapp
```

Comment:
Do we care that the spacing is inconsistent? 
May be better for alignment, but thought I would ask